include/irr/core/math/aabb2dSIMD.h

```diff
@@ -0,0 +1,169 @@
+#ifndef __IRR_AABB_SIMD_2D_H_INCLUDED__
+#define __IRR_AABB_SIMD_2D_H_INCLUDED__
+
+#include "irr/core/math/irrMath.h"
+#include "irr/core/memory/memory.h"
+#include "irr/core/alloc/AlignedBase.h"
+
+#include <iostream>
+
+#define SIGN_FLIP_MASK_XY	core::vector3du32_SIMD(0x80000000, 0x80000000, 0x00000000, 0x00000000)
+#define SIGN_FLIP_MASK_XYZW core::vector3du32_SIMD(0x80000000, 0x80000000, 0x80000000, 0x80000000)
+
+
+namespace irr { namespace core {
+
+class aabb2dSIMDf
+{
+public:
+	inline aabb2dSIMDf()
+		: box( FLT_MAX, FLT_MAX, FLT_MAX, FLT_MAX ) {};
+
+	inline aabb2dSIMDf(const core::vectorSIMDf& _box)
+		: box(_box ^ SIGN_FLIP_MASK_XY) {};
+
+	inline void addBox(const aabb2dSIMDf& _box)		
+	{ 
+		box = core::max_(box, _box.getBounds()); 
+	}
+
+	inline void addPoint(const core::vectorSIMDf& point)	
+	{
+		addBox(point); 
+	}
+
+	inline core::vectorSIMDf getBounds() const
+	{
+		return box ^ SIGN_FLIP_MASK_XY;
+	}
+
+
+	//! Resets the bounding box to a one-point box.
+	/** \param x X coord of the point.
+	\param y Y coord of the point. */
+	inline void reset(float x, float y)
+	{
+		box = core::vectorSIMDf(-x, -y, x, y);
+	}
+
+	//! Resets the bounding box.
+	/** \param initValue New box to set this one to. */
+	inline void reset(const aabb2dSIMDf& initValue)
+	{
+		box = initValue.box;
+	}
+
+	//! Resets the bounding box to a one-point box.
+	/** \param initValue New point. */
+	//void reset(const aabb2dSIMDf& initValue);
+
+	//! Get center of the bounding box
+	/** \return Center of the bounding box. */
+	inline core::vectorSIMDf getCenter() const
+	{
+		core::vectorSIMDf minToCenterVec = (box + box.zwzw()) * 0.5f;
+		return ((box ^ SIGN_FLIP_MASK_XY) + minToCenterVec);
+	}
+
+	//! Get extent of the box (maximal distance of two points in the box)
+	/** \return Extent of the bounding box. */
+	core::vectorSIMDf getExtent() const
+	{
+		return box.zwzw() + box;
```

Comment:
I think its important to have zeros in the zw output coordinate so save to a temporary and call `makeSafe2D` before returning